builtInFunctions/esdtNFTCreate.go

```diff
@@ -141,7 +150,7 @@ func (e *esdtNFTCreate) ProcessBuiltinFunction(
 	}
 
 	tokenID := vmInput.Arguments[0]
-	err = e.rolesHandler.CheckAllowedToExecute(accountWithRoles, vmInput.Arguments[0], []byte(core.ESDTRoleNFTCreate))
+	err = e.isAllowedToExecute(accountWithRoles, vmInput.Arguments[0], []byte(core.ESDTRoleNFTCreate))
```

Comment:
if integrated into rolesHandler - there would be no need for the baseCrossChain checker - everything can be in the roles checker only. It is better from architecture point of view.

Reply:
Right, that's a good suggestion. Refactored.